src/main/java/de/tum/cit/aet/artemis/core/security/passkey/ArtemisPasskeyWebAuthnConfigurer.java

```diff
@@ -0,0 +1,165 @@
+            PublicKeyCredentialCreationOptionsRepository publicKeyCredentialCreationOptionsRepository,
+            PublicKeyCredentialRequestOptionsRepository publicKeyCredentialRequestOptionsRepository) {
+        this.converter = converter;
+        this.jwtCookieService = jwtCookieService;
+        this.userRepository = userRepository;
+        this.publicKeyCredentialUserEntityRepository = publicKeyCredentialUserEntityRepository;
+        this.userCredentialRepository = userCredentialRepository;
+        this.publicKeyCredentialCreationOptionsRepository = publicKeyCredentialCreationOptionsRepository;
+        this.publicKeyCredentialRequestOptionsRepository = publicKeyCredentialRequestOptionsRepository;
+    }
+
+    /**
+     * Validates the configuration for allowed origins when passkey authentication is enabled.
+     * <p>
+     * This method ensures that the server URL and port are correctly configured for WebAuthn
+     * when passkey authentication is enabled. If the configuration is invalid, an exception is thrown.
+     * </p>
+     *
+     * @throws IllegalStateException if the server URL configuration is invalid
+     */
+    @PostConstruct
+    public void validatePasskeyAllowedOriginConfiguration() {
+        if (!passkeyEnabled) {
+            return;
+        }
+
+        try {
+            URL clientUrlToRegisterPasskey = new URI(serverUrl).toURL();
+            URL clientUrlToAuthenticateWithPasskey = new URI(serverUrl + ":" + port).toURL();
+
+            relyingPartyId = clientUrlToRegisterPasskey.getHost();
+
+            // Configure allowed origins
+            allowedOrigins.add(clientUrlToRegisterPasskey.toString());
+            allowedOrigins.add(clientUrlToAuthenticateWithPasskey.toString());
+
+            // Add Android APK key hashes if configured
+            addAndroidApkKeyHashes();
+
+            log.info("WebAuthn passkey authentication enabled with RP ID: {}", relyingPartyId);
+            log.info("Allowed origins: {}", allowedOrigins);
+        }
+        catch (URISyntaxException | MalformedURLException e) {
+            throw new IllegalStateException("Invalid server URL configuration for WebAuthn: " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Adds Android APK key hashes to allowed origins if configured.
+     */
+    private void addAndroidApkKeyHashes() {
+        if (androidSha256CertFingerprintRelease != null) {
+            String hash = AndroidApkKeyHashUtil.getHashFromFingerprint(androidSha256CertFingerprintRelease);
+            allowedOrigins.add(hash);
+            log.info("Added Android release APK key hash: {}", hash);
+        }
+
+        if (androidSha256CertFingerprintDebug != null) {
+            String hash = AndroidApkKeyHashUtil.getHashFromFingerprint(androidSha256CertFingerprintDebug);
+            allowedOrigins.add(hash);
+            log.info("Added Android debug APK key hash: {}", hash);
```

Comment:
I do also think we should change the log level to `warn`
```suggestion
            log.warn("Added Android debug APK key hash: {}", hash);
```